### DISCUSSION TITLE
Custom Metadata Columns in Detail List? - Microsoft Search

### DISCUSSION THREAD

**Discussion by nattomatto** [microsoft-search/pnp-modern-search#2844] [Q&A] [answered]
I feel I'm missing a piece of documentation...  I have custom metadata documents that I have working as refinable strings, and working on the SharePoint Search side.  

What I'm having trouble is to get that data visible in the Microsoft Search side.  Is there a trick to this?  I don't see the columns alliable in the selected fields selection.  I have typed them in directly, but I don't see any data...

I have tried it as resource.[CustomFieldName], resource.fields.[CustomFieldName],RefinableString06...  I'm a bit lost....

What is making my headache worse is that the refinable string works to filter in the Filter web part, and shows proper data.  Where am I going wrong?

**Comment by wobba** [comment, marked as answer]
Use the debug display template as that will list the data structure of what is returned. Then you have the object paths matching your data.

**Reply by nattomatto** [reply]
Thanks for the reply.  Those fields shown in the debug view are the only ones available?  That would mean that Microsoft Search would not be able to use any of the custom metadata fields that are associated with content types.

**Reply by wobba** [reply]
If you request them, yes. I recommend using the Graph Explorer and look at the docs to see what is possible. Features in beta require the setting to enable the beta endpoint for Graph.

**Reply by nattomatto** [reply]
I'm quite familiar with the graph API.  How I would normally do this would be to call would be to use https://graph.microsoft.com/v1.0/sites/[siteID]/lists/[listID]/items?$expand=fields.  This gives me the output that I need.  I didn't see a spot where I could change the call, within the UI, would that be a code update?

**Reply by nattomatto** [reply]
Ok, I figured it out...  I added the property to the list manually, then saved and refreshed!  Lots of data now.  Very helpful!  Thank you very much!

**Reply by wobba** [reply]
@nattomatto I'm talking about the Graph Search API in the explorer - https://learn.microsoft.com/en-us/graph/api/resources/search-api-overview?view=graph-rest-1.0
